src/client.rs

```diff
@@ -79,7 +91,7 @@ impl SigningFederationClient<MatrixConnector> {
         let connector = MatrixConnector::with_default_resolver()?;
 
         Ok(SigningFederationClient {
-            client: Client::builder().build(connector),
+            client: Client::builder(TokioExecutor::new()).build(connector),
```

Comment:
Matches the [`hyper-util` example](https://github.com/hyperium/hyper-util/blob/46826ea75836852fac53ff075a12cba7e290946e/src/client/legacy/client.rs#L203)